### ISSUE TITLE
Missing key(s) in state_dict: "decoder.13.conv.weight", "decoder.13.conv.bias".

### ISSUE THREAD

**Issue by LIUTIANKUO** [nianticlabs/monodepth2#259]
hi, thank you for your great framework!!
I want to reproduce your code with different scales used in the loss. 
I tried 3 scales. Training is correct. But when I evaluate depth, there is a problem showing that 
(Missing key(s) in state_dict: "decoder.13.conv.weight", "decoder.13.conv.bias".) 
This question bothers me, could you give me some advice?

**Comment by mdfirman** [comment]
The problem here, I think, is that you have modified the training code to use different scales, but the evaluate_depth.py doesn't take any of these changes into account. So on evaluation it is trying to build the 'full' network still.

Look at this line here in evaluate_depth.py:

https://github.com/nianticlabs/monodepth2/blob/master/evaluate_depth.py#L90

You might want to try modifying it from

    depth_decoder = networks.DepthDecoder(encoder.num_ch_enc)

to 

    depth_decoder = networks.DepthDecoder(encoder.num_ch_enc, scales=[0, 1, 2])


...or whatever scales you have trained with.

I hope this helps!

**Comment by mdfirman** [comment]
Once you have managed to evaluate please do report the numbers here as we would be interested to see them! Thanks.

**Comment by LIUTIANKUO** [comment]
Thank you for your reply!! Your advice really helps me!
Considering the training time and monodepth2's great efficiency, I only train 1 and 2 scales with 1 epoch, and the result is a little worse than 4 scales

1. scale
abs_rel |   sq_rel |     rmse | rmse_log |       a1 |       a2 |       a3 |
0.140       0.917       5.396    0.214        0.806      0.944   0.981

2. scales 
abs_rel |   sq_rel |     rmse | rmse_log |       a1 |       a2 |       a3 | 
0.154    1.177         5.448     0.219         0.801    0.939    0.979
